Feature: Demo Feedback Theory of Change view enhancements

MalloryLeeWong opened the pull request:
### Changes
- https://github.com/Blue-Marine-Foundation/Maerl/issues/153
- Added (copy pasted) shadcn Tooltip component and installed the radix UI dependency 
- Created constant `logframeText` for commonly used text 

### Screenshots

#### ToC page 
<img width="1677" alt="Screenshot 2025-02-17 at 4 46 28 PM" src="https://github.com/user-attachments/assets/604fc29f-3e4c-4163-aac1-3dfa5d3c320e" />

<img width="1421" alt="Screenshot 2025-02-17 at 4 47 43 PM" src="https://github.com/user-attachments/assets/36ee1d90-f603-409c-87d1-4d8b73bb40cc" />

<img width="1439" alt="Screenshot 2025-02-17 at 4 47 50 PM" src="https://github.com/user-attachments/assets/b293d647-2ad1-4154-8a02-7a6167353799" />

<img width="1541" alt="Screenshot 2025-02-17 at 4 45 04 PM" src="https://github.com/user-attachments/assets/28c21d3a-073c-4cac-9dcf-cfbe6d405c62" />

<img width="1461" alt="Screenshot 2025-02-17 at 4 45 09 PM" src="https://github.com/user-attachments/assets/004046ea-e83e-4558-9c6e-da99518a8476" />

<img width="1433" alt="Screenshot 2025-02-17 at 4 45 13 PM" src="https://github.com/user-attachments/assets/6253042d-4bf4-458e-bb86-165e1d2ffa3e" />

#### Logframe page
Handled empty states when the user has not yet added any Outcome Indicators or Outputs:
<img width="1657" alt="Screenshot 2025-02-17 at 4 44 50 PM" src="https://github.com/user-attachments/assets/f3c58886-c1c9-4a6d-b030-7912f6e0a639" />

Tooltips added here too
<img width="1629" alt="Screenshot 2025-02-17 at 4 45 32 PM" src="https://github.com/user-attachments/assets/dc8446f5-e607-47f9-8a48-21a9140d99ac" />

samjamead reviewed (changes requested):
Yep brill – this is exactly what I had imagined after I had read your summary of the feedback from the demo! Perfect. 

I just caught one thing – I followed your flow outlined on the theory of change page "to add outputs head to the logframe page", and then when poking the empty state of the logframe, I found that the bottom message persisted even after I had added an outcome measurable: 

![Screenshot 2025-02-19 at 12 48 33](https://github.com/user-attachments/assets/a702c884-19d2-4653-8aa7-6f04a3274946)

From a quick look in `@/components/logframe/outputs-container.tsx` I couldn't immediately see why this would happen, because it seems like your ternary should work... It persists on refresh so I don't think it's a state issue. 

Just to check on this – does Sophie want Ouputs to still be children of Outcome Indicators? I thought I remembered them wanting to drop this relationship, but you have spoken to them more recently than me! 

I will delete the output measurable I added so that you can retest the empty state

MalloryLeeWong commented:
> Yep brill – this is exactly what I had imagined after I had read your summary of the feedback from the demo! Perfect.
> 
> I just caught one thing – I followed your flow outlined on the theory of change page "to add outputs head to the logframe page", and then when poking the empty state of the logframe, I found that the bottom message persisted even after I had added an outcome measurable:
> 
> ![Screenshot 2025-02-19 at 12 48 33](https://private-user-images.githubusercontent.com/3154415/414725991-a702c884-19d2-4653-8aa7-6f04a3274946.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzk5NzUxMzksIm5iZiI6MTczOTk3NDgzOSwicGF0aCI6Ii8zMTU0NDE1LzQxNDcyNTk5MS1hNzAyYzg4NC0xOWQyLTQ2NTMtOGFhNy02ZjA0YTMyNzQ5NDYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDIxOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAyMTlUMTQyMDM5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZWJmNDQ0MDczMTdkY2I3YzgyMjIxMjExZmE3NmFmM2ZjZjY3MzkxMTUwMzYyNTZkN2E2YjRjOTQwYmNlN2EyZiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.37NXKqBwNM1YSA9OeWcwXqRTKNaB6em-4WIFNJ2Ai3c)
> 
> From a quick look in `@/components/logframe/outputs-container.tsx` I couldn't immediately see why this would happen, because it seems like your ternary should work... It persists on refresh so I don't think it's a state issue.
> 
> Just to check on this – does Sophie want Ouputs to still be children of Outcome Indicators? I thought I remembered them wanting to drop this relationship, but you have spoken to them more recently than me!
> 
> I will delete the output measurable I added so that you can retest the empty state

Thanks @samjamead , great catch! Since the existing pop-up form to create or edit an Output requires selecting an Outcome Measurable, I had created this empty state flow based on that relationship where Outputs are children of Outcome Indicators. When I met with Sophie and Suneha during the demo they didn't comment on what they wanted the relationship between these to be. 

As the next step, would you recommend that I fix the bug and keep the existing relationship, reach out to confirm with Sophie, or drop this relationship and make it so Outputs are not children of Outcome Indicators? If you recall them saying they didn't want that nested relationship anymore I can look at how much effort it would be to change it and if it's a small lift I can go ahead and make the change here.

samjamead commented:
Thanks for the extra details – ok, I think it's probably easiest for us to ditch the mandatory relationship – we can still allow the user to nest an `output` under an `outcome measurable` when they create new ones, but we won't require it  👍

MalloryLeeWong commented:
> Thanks for the extra details – ok, I think it's probably easiest for us to ditch the mandatory relationship – we can still allow the user to nest an `output` under an `outcome measurable` when they create new ones, but we won't require it 👍

@samjamead Thank you, great. I pushed changes that remove the mandatory relationship. So now when the user creates or edits  their first or any subsequent Outputs, selecting an Outcome Measurable is optional now.

### Screenshots
Here's how it looks now on the ToC page. The user can directly add their first output on the ToC page without going to the Logframe and without selecting an Outcome Measurable.
<img width="1465" alt="Screenshot 2025-02-19 at 8 49 46 PM" src="https://github.com/user-attachments/assets/6d7f1b07-0c3c-4aba-a0e5-71befae49f00" />
<img width="655" alt="Screenshot 2025-02-19 at 8 52 54 PM" src="https://github.com/user-attachments/assets/99b14910-f66e-42e9-a4a0-9e3c567b0d39" />

And now on the Logframe page, if they haven't created an Output yet, there is just an "add output" button at the bottom under the other cards:
<img width="1415" alt="Screenshot 2025-02-19 at 8 50 52 PM" src="https://github.com/user-attachments/assets/782e39ad-b73b-4d14-961e-ee14d926333d" />

samjamead reviewed (approved):
Brill thanks Mallory. I spotted a couple of small things when I reviewed last night but rather than add them as notes they were easier to just fix quickly and commit. 

Thanks again for your work on this PR, it's a real beast and the two main pages affected look awesome now! 

Clear to merge into production  👍

MalloryLeeWong commented:
> Brill thanks Mallory. I spotted a couple of small things when I reviewed last night but rather than add them as notes they were easier to just fix quickly and commit.
> 
> Thanks again for your work on this PR, it's a real beast and the two main pages affected look awesome now!
> 
> Clear to merge into production 👍

Thank @samjamead , appreciate your additions and feedback! Will go ahead and merge this PR into our development branch `feat/theory-of-change-and-logframe-view-enhancements`, so we can test and release that to production next.
